lib/modules/manager/gradle/extract/consistent-versions-plugin.ts

```diff
@@ -0,0 +1,187 @@
+
+  // For each exact dep in props file
+  for (const [propDep, versionAndPosition] of propsFileExactMap) {
+    if (lockFileMap.has(propDep)) {
+      const newDep: Record<string, any> = {
+        managerData: {
+          packageFile: VERSIONS_PROPS,
+          fileReplacePosition: versionAndPosition.filePos,
+        },
+        packageName: propDep,
+        currentValue: versionAndPosition.version,
+        currentVersion: versionAndPosition.version,
+        lockedVersion: lockFileMap.get(propDep),
+      } as PackageDependency<GradleManagerData>;
+      extractedDeps.push(newDep);
+    }
+  }
+
+  // For each regular expression (group) dep in props file
+  for (const [propDepRegEx, propVerAndPos] of propsFileRegexMap) {
+    for (const [exactDep, exactVer] of lockFileMap) {
+      if (propDepRegEx.test(exactDep)) {
+        const newDep: Record<string, any> = {
+          managerData: {
+            packageFile: VERSIONS_PROPS,
+            fileReplacePosition: propVerAndPos.filePos,
+          },
+          depName: exactDep,
+          currentValue: propVerAndPos.version,
+          currentVersion: propVerAndPos.version,
+          lockedVersion: exactVer,
+          groupName: regexToGroupString(propDepRegEx),
+        } as PackageDependency<GradleManagerData>;
+        extractedDeps.push(newDep);
+      }
+    }
+  }
+  return extractedDeps;
+}
+
+//----------------------------------
+// Private utility functions below
+//----------------------------------
+
+// Translate glob syntax to a regex that does the same
+function globToRegex(depName: string): RegExp {
+  return regEx(
+    depName
+      .replaceAll('*', '_WC_CHAR_')
+      .replace(/[/\-\\^$*+?.()|[\]{}]/g, '\\$&')
+      .replaceAll('_WC_CHAR_', '[^:]*')
+  );
+}
+
+// Translate the regex from versions.props into a group name used in branch names etc
+function regexToGroupString(regExp: RegExp): string {
+  return regExp.source.replaceAll('[^:]*', '-').replaceAll('\\.', '.');
+}
+
+interface VersionWithPosition {
+  version?: string;
+  filePos?: number;
+}
+
+/**
+ * Parses `versions.lock`
+ */
+function parseLockFile(input: string): Map<string, string> {
+  const lockLineRegex = regEx(
+    `^(?<depName>[^:]+:[^:]+):(?<lockVersion>[^ ]+) \\(\\d+ constraints: [0-9a-f]+\\)$`
+  );
+
+  const depVerMap = new Map<string, string>();
+  for (const line of input.split(newlineRegex)) {
+    const lineMatch = lockLineRegex.exec(line);
+    if (lineMatch?.groups) {
+      const { depName, lockVersion } = lineMatch.groups;
+      depVerMap.set(depName, lockVersion);
+    }
+  }
+  logger.trace(
+    `Found ${depVerMap.size} locked dependencies in ${VERSIONS_LOCK}.`
+  );
+  return depVerMap;
+}
+
+/**
+ * Parses `versions.props`
+ */
+function parsePropsFile(
+  input: string
+): [Map<string, VersionWithPosition>, Map<RegExp, VersionWithPosition>] {
+  const propsLineRegex = regEx(
+    `^(?<depName>[^:]+:[^=]+?) *= *(?<propsVersion>.*)$`
```

Comment:
This pattern is extremely relaxed. It also matches, e.g. `some garbage text with : and also = for testing`

Make sure to reuse `isDependencyString`, `versionLikeSubstring` and probably `artifactRegex` from utils at some point to ensure no garbage is parsed as a potential dependency.

Reply:
Added extra validation to parsing

Reply:
As of now it still matches: `asdf:qwer:asdf:qwer = 234`, which is for sure invalid. At the same time, it won't match `org.assertj* = 234` - I assume this would be valid too, since `*` can occur at any position?